draft-ietf-quic-http.md

```diff
@@ -963,20 +963,20 @@ for each header field.
 
 ### Request Cancellation
 
-Either client or server can cancel requests by aborting the stream (QUIC
-RESET_STREAM and/or STOP_SENDING frames, as appropriate) with an error code of
+Clients can cancel requests by aborting the stream (QUIC RESET_STREAM and/or
+STOP_SENDING frames, as appropriate) with an error code of
 HTTP_REQUEST_CANCELLED ({{http-error-codes}}).  When the client cancels a
 response, it indicates that this response is no longer of interest.
 Implementations SHOULD cancel requests by aborting both directions of a stream.
```

Comment:
Missing from either this paragraph or the next:  A server that receives a STOP_SENDING with the HTTP_REQUEST_CANCELLED error code MAY use the HTTP_REQUEST_REJECTED error code in response to indicate that no processing occurred.

Reply:
Wouldn't that defeat the purpose of this PR? If a server receives STOP_SENDING with HTTP_REQUEST_CANCELLED, it doesn't reject the request, it acts on the client's cancellation. In that case I think it would make more sense for the server to use HTTP_REQUEST_CANCELLED.

Reply:
From `-transport`:
> An endpoint SHOULD copy the error code from the STOP_SENDING frame, but MAY use any application error code.

_CANCELLED would be the default behavior; this text merely says that _REJECTED is also acceptable.  (And the transport draft says that the server MAY use any error code, so this is really just a restatement.)

The subtle difference would be an indication of whether the cancelled request had begun to process (and therefore might have had side-effects) before the cancellation took place.

Reply:
Right. The trick here is that a cancelled request can't be automatically retried.  A rejected one can be.  So using _REJECTED is preferable.

Reply:
Ah, I see. I agree. Hmmm not sure how to clarify the text so others aren't confused the same way I was though.